rs/nns/governance/src/proposals/install_code.rs

```diff
@@ -141,6 +145,37 @@ impl CallCanister for InstallCode {
     }
 }
 
+impl CallCanister for InstallCode {
+    fn canister_and_function(&self) -> Result<(CanisterId, &str), GovernanceError> {
```

Comment:
Maybe add a triple slash comment that says something like

> Most canisters are upgraded indirectly via root. In such cases, we call root's change_nns_canister method. The exception is when root is to be upgraded. In that case, upgrades are instead done via lifeline's upgrade_root method.

Reply:
Added comment to the if clause. IMHO comments on the method should describe its behavior (returns canister id and the function name for proposal execution), while the comment suggested here seems to be describing an implementation detail.